zeta-chain/node
# fix(`rpc`): reduce Ethereum rpc API and improve websocket API

# Description

- Keep only the namespaces available on regular Ethereum rpcs. Disable `eth_sendTransaction`, `eth_sign` and `eth_signTypedData`
- Set message read limit to 32 MB

To test:
- Start smoke tests
- Check APIs

Before:

```
curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"rpc_modules","id":1}' http://localhost:9545
{"jsonrpc":"2.0","id":1,"result":{"debug":"1.0","eth":"1.0","miner":"1.0","net":"1.0","personal":"1.0","rpc":"1.0","txpool":"1.0","web3":"1.0"}}
```

After

```
curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"rpc_modules","id":1}' http://localhost:9545
{"jsonrpc":"2.0","id":1,"result":{"eth":"1.0","net":"1.0","rpc":"1.0","web3":"1.0"}}

curl -X POST --data '{"jsonrpc":"2.0","method":"personal_sign","params":["0x48656c6c6f", "0x604BB0f27A24374f1a784C458aFB9210CaAA2EDA", "password"],"id":1}' -H "Content-Type: application/json" http://localhost:9545/
{"jsonrpc":"2.0","id":1,"error":{"code":-32601,"message":"the method personal_sign does not exist/is not available"}}

curl -X POST --data '{"jsonrpc":"2.0","method":"eth_sendTransaction","params":[{}],"id":1}' -H "Content-Type: application/json" http://localhost:9545

curl -X POST --data '{"jsonrpc":"2.0","method":"eth_sign","params":["0x604BB0f27A24374f1a784C458aFB9210CaAA2EDA", "0x48656c6c6f"],"id":1}' -H "Content-Type: application/json" http://localhost:9545

curl -X POST --data '{"jsonrpc":"2.0","method":"eth_signTypedData","params":["0x604BB0f27A24374f1a784C458aFB9210CaAA2EDA", {}],"id":1}' -H "Content-Type: application/json" http://localhost:9545
```

## Type of change


# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works
